<a href="https://colab.research.google.com/github/Suwannaket1/python_dataviz2024/blob/main/Exam_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**นางสาวสุวรรณเกตุ สุมาลี รหัสนักศึกษา 653020578-2**

# ให้สืบหาฆาตรกร ในคดี ฆาตรกรรม (murder) ที่เกิดขึ้นในวันที่ 15 มกราคม 2018 ที่เมือง SQL City โดยเริ่มจากการตรวจสอบ ข้อมูลคดี (crime_scene_report.csv)

หมายเหตุ

person.csv ข้อมูลประชากร

interview.csv ข้อมูลการสอบสวน

income.csv ข้อมูลรายได้

get_fit_now_member.csv ข้อมูลสมาชิก ฟิตเนส Get Fit Now

get_fit_now_check_in.csv ข้อมูลการเข้าใช้ ฟิตเนส Get Fit Now

facebook_event_checkin.csv ข้อมูลการเช็คอิน Facebook

driver_license.csv ข้อมูลใบขับขี่

crime_scene_report.csv ข้อมูลคดี

#นำเข้าข้อมูล

In [1]:
import pandas as pd

In [2]:
import os

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
path_to_person = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/person.csv'
path_to_interview = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/interview.csv'
path_to_income = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/income.csv'
path_to_get_fit_now_member = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/get_fit_now_member.csv'
path_to_get_fit_now_check = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/get_fit_now_check_in.csv'
path_to_facebook_event_checkin = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/facebook_event_checkin.csv'
path_driver_license = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/drivers_license.csv'
path_to_crime_scene_report = '/content/drive/MyDrive/data_viz_2024_DATA/Exam_2/crime_scene_report.csv'

In [5]:
Person_df = pd.read_csv(path_to_person)
Interview_df = pd.read_csv(path_to_interview)
Income_df = pd.read_csv(path_to_income)
GetFitNowMember_df = pd.read_csv(path_to_get_fit_now_member)
GetFitNowCheck_df = pd.read_csv(path_to_get_fit_now_check)
FacebookEventCheckin_df = pd.read_csv(path_to_facebook_event_checkin)
DriverLicense_df = pd.read_csv(path_driver_license)
CrimeSceneReport_df = pd.read_csv(path_to_crime_scene_report)

#รายละเอียดคดี

In [6]:
#ดูข้อมูลคดี
CrimeSceneReport_df

date     type                                        description  \
0     20180115  robbery  A Man Dressed as Spider-Man Is on a Robbery Spree   
1     20180115   murder                  Life? Dont talk to me about life.   
2     20180115   murder  Mama, I killed a man, put a gun against his he...   
3     20180215   murder                         REDACTED REDACTED REDACTED   
4     20180215   murder  Someone killed the guard! He took an arrow to ...   
...        ...      ...                                                ...   
1223  20180430  bribery                                                 \n   
1224  20180430    fraud                  ‘Why not?’ said the March Hare.\n   
1225  20180430  assault                                                 \n   
1226  20180501  assault  be NO mistake about it: it was neither more no...   
1227  20180115   murder  Security footage shows that there were 2 witne...   

              city  
0              NYC  
1           Albany  
2             Reno  
3         SQL City  
4         SQL City  
...            ...  
1223  Garden Grove  
1224         Houma  
1225       Fontana  
1226       Trenton  
1227      SQL City  

[1228 rows x 4 columns]

In [7]:
CrimeSceneReport_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1228 entries, 0 to 1227
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   date         1228 non-null   int64 
 1   type         1228 non-null   object
 2   description  1228 non-null   object
 3   city         1228 non-null   object
dtypes: int64(1), object(3)
memory usage: 38.5+ KB


In [8]:
# แปลง Dtype คอลัมน์ date เป็น date
CrimeSceneReport_df['date'] = pd.to_datetime(CrimeSceneReport_df['date'], format='%Y%m%d')

# ปรับรูปแบบวันที่ให้เป็น DD-MM-YYYY
CrimeSceneReport_df['date'] = CrimeSceneReport_df['date'].dt.strftime('%d %m %Y')

CrimeSceneReport_df

date     type                                        description  \
0     15 01 2018  robbery  A Man Dressed as Spider-Man Is on a Robbery Spree   
1     15 01 2018   murder                  Life? Dont talk to me about life.   
2     15 01 2018   murder  Mama, I killed a man, put a gun against his he...   
3     15 02 2018   murder                         REDACTED REDACTED REDACTED   
4     15 02 2018   murder  Someone killed the guard! He took an arrow to ...   
...          ...      ...                                                ...   
1223  30 04 2018  bribery                                                 \n   
1224  30 04 2018    fraud                  ‘Why not?’ said the March Hare.\n   
1225  30 04 2018  assault                                                 \n   
1226  01 05 2018  assault  be NO mistake about it: it was neither more no...   
1227  15 01 2018   murder  Security footage shows that there were 2 witne...   

              city  
0              NYC  
1           Albany  
2             Reno  
3         SQL City  
4         SQL City  
...            ...  
1223  Garden Grove  
1224         Houma  
1225       Fontana  
1226       Trenton  
1227      SQL City  

[1228 rows x 4 columns]

In [9]:
#ปรับเดือนให้เป็นเดือนภาษาไทย
thai_month = {"01": "มกราคม", "02": "กุมภาพันธ์", "03": "มีนาคม", "04": "เมษายน",
              "05": "พฤษภาคม", "06": "มิถุนายน", "07": "กรกฎาคม", "08": "สิงหาคม",
              "09": "กันยายน", "10": "ตุลาคม", "11": "พฤศจิกายน", "12": "ธันวาคม"}

CrimeSceneReport_df['date'] = CrimeSceneReport_df['date'].apply(lambda x: x.split()[0] + " " + thai_month[x.split()[1]] + " " + x.split()[2])

CrimeSceneReport_df

date     type  \
0         15 มกราคม 2018  robbery   
1         15 มกราคม 2018   murder   
2         15 มกราคม 2018   murder   
3     15 กุมภาพันธ์ 2018   murder   
4     15 กุมภาพันธ์ 2018   murder   
...                  ...      ...   
1223      30 เมษายน 2018  bribery   
1224      30 เมษายน 2018    fraud   
1225      30 เมษายน 2018  assault   
1226     01 พฤษภาคม 2018  assault   
1227      15 มกราคม 2018   murder   

                                            description          city  
0     A Man Dressed as Spider-Man Is on a Robbery Spree           NYC  
1                     Life? Dont talk to me about life.        Albany  
2     Mama, I killed a man, put a gun against his he...          Reno  
3                            REDACTED REDACTED REDACTED      SQL City  
4     Someone killed the guard! He took an arrow to ...      SQL City  
...                                                 ...           ...  
1223                                                 \n  Garden Grove  
1224                  ‘Why not?’ said the March Hare.\n         Houma  
1225                                                 \n       Fontana  
1226  be NO mistake about it: it was neither more no...       Trenton  
1227  Security footage shows that there were 2 witne...      SQL City  

[1228 rows x 4 columns]

In [10]:
# กรองข้อมูลคดีฆาตรกรรมในวันที่ 15 มกราคม 2018 ที่เมือง SQL City
Crime_df = CrimeSceneReport_df[(CrimeSceneReport_df['type'] == 'murder') &
                               (CrimeSceneReport_df['date'] == '15 มกราคม 2018') &
                               (CrimeSceneReport_df['city'] == 'SQL City')]
Crime_df

date    type  \
1227  15 มกราคม 2018  murder   

                                            description      city  
1227  Security footage shows that there were 2 witne...  SQL City

In [11]:
print(Crime_df['description'].iloc[0])

Security footage shows that there were 2 witnesses. The first witness lives at the last house on "Northwestern Dr". The second witness, named Annabel, lives somewhere on "Franklin Ave".


ข้อมูลที่ 1

คดี : ฆาตรกรรม

วันเกิดเหตุ : 15 มกราคม 2018

สถานที่เกิดเหตุ : เมือง SQL City

หลักฐาน : ภาพจากกล้องวงจรปิด ที่ถ่ายติดพยาน 2 คน โดยที่พยานคนแรกเขาอาศัยอยู่บ้านหลังสุดท้าย ที่ถนน Northwestern Dr และพยานคนที่สอง ชื่อ Annabel อาศัยอยู่ซักแห่งที่ถนน Franklin Ave

#ดูข้อมูลพยาน

In [12]:
#ดูข้อมูลบุคคลที่เกี่ยวข้อง
Person_df

id                 name  license_id  address_number  \
0      10000   Christoper Peteuil      993845             624   
1      10007  Kourtney Calderwood      861794            2791   
2      10010            Muoi Cary      385336             741   
3      10016          Era Moselle      431897            1987   
4      10025         Trena Hornby      550890             276   
...      ...                  ...         ...             ...   
10006  99936          Luba Benser      274427             680   
10007  99941      Roxana Mckimley      975942            1613   
10008  99965      Cherie Zeimantz      287627            3661   
10009  99982          Allen Cruse      251350            3126   
10010  99990         Vance Hunten      830407            3056   

      address_street_name        ssn  
0            Bankhall Ave  747714076  
1           Gustavus Blvd  477972044  
2         Northwestern Dr  828638512  
3           Wood Glade St  614621061  
4           Daws Hill Way  223877684  
...                   ...        ...  
10006        Carnage Blvd  685095054  
10007             Gate St  512136801  
10008       The Water Ave  362877324  
10009           N Jean Dr  348734531  
10010       Lancefield St  896677562  

[10011 rows x 6 columns]

##พยานคนที่ 1 อาศัยอยู่บ้านหลังสุดท้าย ที่ถนน Northwestern Dr

In [13]:
witness_1 = Person_df[Person_df['address_street_name'] == 'Northwestern Dr']
witness_1

id                  name  license_id  address_number  \
2     10010             Muoi Cary      385336             741   
282   12711        Norman Apolito      667757             599   
499   14887        Morty Schapiro      118009            4919   
528   15171          Weldon Penso      336999             311   
811   17729        Lasonya Wildey      439686            3824   
904   18376              Josh Shi      653712            1091   
1021  19420           Cody Schiel      890431            3524   
1338  22239         Dusty Sigafus      710517            1125   
1426  23044          Val Portlock      924989            3143   
1522  23960  Kristopher Lagerberg      658777            1392   
1686  25615         Olevia Morena      388477             812   
1812  26758      Angelena Billman      942074             629   
1981  28360        Rashad Cascone      924517            3212   
2131  29608       Willetta Lurvey      800568            1028   
2535  33415         Homer Simpson      108392             981   
2620  34155      Allyson Lazenson      814209            2840   
2635  34352    Vincenza Burkhardt      436470            2512   
2871  36378        Freddie Ellzey      267882            3449   
3237  39688            Abe Roeker      537445             689   
3316  40336       Fredia Billeter      251082             404   
3368  40760          Renata Faver      976650             745   
3743  44004           Alison Eska      327797            2951   
3795  44542        Ivory Mcgrevey      873842             897   
4595  51114     Christena Saffell      814284            3055   
4643  51552           Bonita Knie      764124            1726   
4806  53076           Boris Bijou      664914            3327   
4871  53699      Christian Pagano      628433            2585   
4886  53890        Sophie Tiberio      957671            3755   
4956  54564         Sammie Sengun      217286            2590   
5024  55130        Shiloh Huntoon      472118             951   
5450  59307          Cyril Yongue      753524            2903   
6476  68690            Yer Modest      852404            3046   
6698  70485     Andera Kostelecky      354822            1252   
6865  71924        Laurie Scronce      774116            1418   
6877  72076     Courtney Bordeaux      324614            1707   
7040  73368        Torie Thalmann      773862            3697   
7296  75484          Del Tacderen      447887             607   
7888  80921       Donovan Yackley      222395            2386   
7902  81052     Calvin Czelusniak      471483            1516   
7993  81885         Kurtis Masuda      934201            2108   
8369  85280             Donn Duin      569122             540   
8608  87456   Leonora Wolfsberger      215868            3483   
8720  88423           Teri Ehrich      510876            1879   
8888  89906       Kinsey Erickson      510019             309   
8939  90397     Dionna Kranwinkle      797260            2678   
9272  93509        Emmitt Aceuedo      916706            3491   
9569  96178           Galina Khuu      450248            2183   
9609  96595         Coretta Cubie      303645            3631   
9841  98593           Jonah Toner      578982            2947   
9991  99826          Ivy Kazarian      994470             892   

     address_street_name        ssn  
2        Northwestern Dr  828638512  
282      Northwestern Dr  778264744  
499      Northwestern Dr  111564949  
528      Northwestern Dr  131379495  
811      Northwestern Dr  917817122  
904      Northwestern Dr  193899001  
1021     Northwestern Dr  947110049  
1338     Northwestern Dr  724386723  
1426     Northwestern Dr  100593316  
1522     Northwestern Dr  492912529  
1686     Northwestern Dr  962831331  
1812     Northwestern Dr  341854756  
1981     Northwestern Dr  838907424  
2131     Northwestern Dr  843751378  
2535     Northwestern Dr  871392019  
2620     Northwestern Dr  991150469  
2635     Northwestern Dr  197913046  
2871     

In [14]:
#ดูบ้านเลขที่หลังสุดท้าย (บ้านเลขที่ที่จำนวนมากที่สุด)
index_max = witness_1['address_number'].idxmax()
witness_1 = witness_1.loc[[index_max]]
witness_1

id            name  license_id  address_number address_street_name  \
499  14887  Morty Schapiro      118009            4919     Northwestern Dr   

           ssn  
499  111564949

**พยานคนที่ 1** คือ

name : Morty Schapiro

id : 14887

ssn : 111564949

## พยานคนที่ 2 ชื่อ Annabel อาศัยอยู่ซักแห่งที่ถนน Franklin Ave

In [15]:
witness_2 = Person_df[Person_df['name'].str.startswith('Annabel')]
witness_2

id                name  license_id  address_number  \
665   16371      Annabel Miller      490173             103   
7601  78354      Annabell Siona      158932             978   
7650  78799  Annabell Droneburg      984316            1944   
8509  86541   Annabell Zwilling      709133            1859   

     address_street_name        ssn  
665         Franklin Ave  318771143  
7601       Whitewater Dr  800278294  
7650        W Natalie Dr  478793500  
8509            Patti Rd  332961158

พยานคนที่ 2 คือ

name : Annabel Miller

id : 16371

ssn : 318771143

#ดูข้อมูลการสอบสวนพยาน

In [16]:
Interview_df

person_id                                         transcript
0         28508                ‘I deny it!’ said the March Hare.\n
1         63713                                                 \n
2         86208      way, and the whole party swam to the shore.\n
3         35267  lessons in here? Why, there’s hardly room for ...
4         33856                                                 \n
...         ...                                                ...
4986      37357  Alice did not wish to offend the Dormouse agai...
4987      10206  time,’ she said, ‘than waste it in asking ridd...
4988      14887  I heard a gunshot and then saw a man run out. ...
4989      16371  I saw the murder happen, and I recognized the ...
4990      67318  I was hired by a woman with a lot of money. I ...

[4991 rows x 2 columns]

##คำให้การของพยานคนที่ 1 (คุณ Morty Schapiro)

In [17]:
Interview_df[Interview_df['person_id'] == 14887].iloc[0]['transcript']

'I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".'

Morty Schapiro ให้การว่า " ฉันได้ยินเสียงปืนแล้วเห็นผู้ชายคนหนึ่งวิ่งออกมา เขาถือกระเป๋า 'Get Fit Now Gym' หมายเลขสมาชิกบนกระเป๋าเริ่มต้นด้วย '48Z' ซึ่งเป็นกระเป๋าที่มีเฉพาะสมาชิกระดับ Gold เท่านั้น ผู้ชายคนนั้นขึ้นรถที่มีป้ายทะเบียนที่มีตัวอักษร 'H42W' "

##คำให้การของพยานคนที่ 2 (คุณ Annabel Miller)

In [18]:
Interview_df[Interview_df['person_id'] == 16371].iloc[0]['transcript']

'I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.'

Annabel Miller ให้การว่า " ฉันเห็นการฆาตกรรมเกิดขึ้นและฉันจำฆาตกรได้จากยิมของฉันเมื่อฉันออกกำลังกายในสัปดาห์ที่แล้วเมื่อวันที่ 9 มกราคม "

#สืบคดีต่อจากคำให้การของพยานคนที่ 1 (คุณ Morty Schapiro)


###ตรวจสอบจากข้อมูลสมาชิกฟิตเนส
โดยจะเริ่มกรองสมาชิก Get Fit Now Gym ที่เป็นสมาชิกระดับ Gold และกรองหมายเลขสมาชิกที่เริ่มต้นด้วย '48Z'

In [19]:
#ดูข้อมูลสมาชิกฟิตเนส
GetFitNowMember_df

id  person_id                name  membership_start_date  \
0    NL318      65076     Everette Koepke               20170926   
1    AOE21      39426        Noe Locascio               20171005   
2    2PN28      63823  Jeromy Heitschmidt               20180215   
3    0YJ24      80651      Waneta Wellard               20171206   
4    3A08L      32858        Mei Bianchin               20170401   
..     ...        ...                 ...                    ...   
179  2V137      41693      Wendell Dulany               20171219   
180  4KB72      79110          Emile Hege               20170522   
181  48Z7A      28819        Joe Germuska               20160305   
182  48Z55      67318       Jeremy Bowers               20160101   
183  90081      16371      Annabel Miller               20160208   

    membership_status  
0                gold  
1             regular  
2              silver  
3                gold  
4              silver  
..                ...  
179            silver  
180           regular  
181              gold  
182              gold  
183              gold  

[184 rows x 5 columns]

In [20]:
evidence1 = GetFitNowMember_df['id'].str.startswith('48Z')
evidence2 = GetFitNowMember_df['membership_status'] == 'gold'

Suspect = GetFitNowMember_df[evidence1 & evidence2]

Suspect

id  person_id           name  membership_start_date membership_status
181  48Z7A      28819   Joe Germuska               20160305              gold
182  48Z55      67318  Jeremy Bowers               20160101              gold

**ผู้ต้องสงสัยคนที่ 1**

name : Joe Germuska

person_id : 28819

member_id : 48Z7A

--------------------

**ผู้ต้องสงสัยคนที่ 2**

name : Jeremy Bowers

person_id : 67318

member_id : 48Z55

###ตรวจสอบจากข้อมูลใบขับขี่
โดยจะกรองจากป้ายทะเบียนที่มีตัวอักษร 'H42W'

In [21]:
#ดูใบขับขี่
DriverLicense_df

id  age  height eye_color hair_color  gender plate_number  \
0      100280   72      57     brown        red    male       P24L4U   
1      100460   63      72     brown      brown  female       XF02T6   
2      101029   62      74     green      green  female       VKY5KR   
3      101198   43      54     amber      brown  female       Y5NZ08   
4      101255   18      79      blue       grey  female       5162Z1   
...       ...  ...     ...       ...        ...     ...          ...   
10002  999923   19      77     amber      black  female       5L0ZI4   
10003  999940   71      61     green      green    male       1B8QN8   
10004  999981   67      69     brown       blue  female       1684K3   
10005  999986   49      58     green       grey    male       F8F64H   
10006  999993   18      63     black      black  female       6UZO2O   

         car_make    car_model  
0           Acura          MDX  
1        Cadillac          SRX  
2           Scion           xB  
3          Nissan        Rogue  
4           Lexus           GS  
...           ...          ...  
10002         GMC  Sierra 3500  
10003  Mitsubishi      Eclipse  
10004  Land Rover          LR2  
10005       Lexus           LS  
10006    Cadillac      DeVille  

[10007 rows x 9 columns]

In [22]:
DriverLicense_df[DriverLicense_df['plate_number'].str.contains('H42W')]

id  age  height eye_color hair_color  gender plate_number  \
915   183779   21      65      blue     blonde  female       H42W0X   
3529  423327   30      70     brown      brown    male       0H42W2   
6240  664760   21      71     black      black    male       4H42WR   

       car_make car_model  
915      Toyota     Prius  
3529  Chevrolet  Spark LS  
6240     Nissan    Altima

In [23]:
Person_df[Person_df['license_id'] == 183779]

id            name  license_id  address_number address_street_name  \
7578  78193  Maxine Whitely      183779             110             Fisk Rd   

            ssn  
7578  137882671

In [24]:
Person_df[Person_df['license_id'] == 423327]

id           name  license_id  address_number    address_street_name  \
6327  67318  Jeremy Bowers      423327             530  Washington Pl, Apt 3A   

            ssn  
6327  871539279

In [25]:
Person_df[Person_df['license_id'] == 664760]

id            name  license_id  address_number address_street_name  \
4664  51739  Tushar Chandra      664760             312              Phi St   

            ssn  
4664  137882671

*สรุป* ผู้ต้องสงสัยจากคำให้การของคุณ Morty Schapiro ชี้ไปที่ Jeremy Bowers

#สืบคดีต่อจากคำให้การของพยานคนที่ 2 (คุณ Annabel Miller)
โดยจะกรองข้อมูลผู้เข้าใช้งานฟิตเนส วันที่ 09 มกราคม 2018

In [26]:
#ดูข้อมูลการเข้าใช้ฟิตเนส
GetFitNowCheck_df

membership_id  check_in_date  check_in_time  check_out_time
0            NL318       20180212            329             365
1            NL318       20170811            469             920
2            NL318       20180429            506             554
3            NL318       20180128            124             759
4            NL318       20171027            418            1019
...            ...            ...            ...             ...
2698         4KB72       20170422           1016            1114
2699         4KB72       20170630            408             885
2700         48Z7A       20180109           1600            1730
2701         48Z55       20180109           1530            1700
2702         90081       20180109           1600            1700

[2703 rows x 4 columns]

In [27]:
# แปลงคอลัมน์ date เป็นวันที่
GetFitNowCheck_df['check_in_date'] = pd.to_datetime(GetFitNowCheck_df['check_in_date'], format='%Y%m%d')

# ปรับรูปแบบวันที่ให้เป็น DD-MM-YYYY
GetFitNowCheck_df['check_in_date'] = GetFitNowCheck_df['check_in_date'].dt.strftime('%d %m %Y')

GetFitNowCheck_df

membership_id check_in_date  check_in_time  check_out_time
0            NL318    12 02 2018            329             365
1            NL318    11 08 2017            469             920
2            NL318    29 04 2018            506             554
3            NL318    28 01 2018            124             759
4            NL318    27 10 2017            418            1019
...            ...           ...            ...             ...
2698         4KB72    22 04 2017           1016            1114
2699         4KB72    30 06 2017            408             885
2700         48Z7A    09 01 2018           1600            1730
2701         48Z55    09 01 2018           1530            1700
2702         90081    09 01 2018           1600            1700

[2703 rows x 4 columns]

In [28]:
#แปลงเดือนให้เป็นเดือนภาษาไทย
thai_month = {"01": "มกราคม", "02": "กุมภาพันธ์", "03": "มีนาคม", "04": "เมษายน",
              "05": "พฤษภาคม", "06": "มิถุนายน", "07": "กรกฎาคม", "08": "สิงหาคม",
              "09": "กันยายน", "10": "ตุลาคม", "11": "พฤศจิกายน", "12": "ธันวาคม"}

GetFitNowCheck_df['check_in_date'] = GetFitNowCheck_df['check_in_date'].apply(lambda x: x.split()[0] + " " + thai_month[x.split()[1]] + " " + x.split()[2])

GetFitNowCheck_df

membership_id       check_in_date  check_in_time  check_out_time
0            NL318  12 กุมภาพันธ์ 2018            329             365
1            NL318     11 สิงหาคม 2017            469             920
2            NL318      29 เมษายน 2018            506             554
3            NL318      28 มกราคม 2018            124             759
4            NL318      27 ตุลาคม 2017            418            1019
...            ...                 ...            ...             ...
2698         4KB72      22 เมษายน 2017           1016            1114
2699         4KB72    30 มิถุนายน 2017            408             885
2700         48Z7A      09 มกราคม 2018           1600            1730
2701         48Z55      09 มกราคม 2018           1530            1700
2702         90081      09 มกราคม 2018           1600            1700

[2703 rows x 4 columns]

In [29]:
GetFitNowCheck_df[GetFitNowCheck_df["check_in_date"] == '09 มกราคม 2018']

membership_id   check_in_date  check_in_time  check_out_time
113          X0643  09 มกราคม 2018            957            1164
889          UK1F2  09 มกราคม 2018            344             518
1329         XTE42  09 มกราคม 2018            486            1124
1446         1AE2H  09 มกราคม 2018            461             944
1703         6LSTG  09 มกราคม 2018            399             515
2137         7MWHJ  09 มกราคม 2018            273             885
2435         GE5Q8  09 มกราคม 2018            367             959
2700         48Z7A  09 มกราคม 2018           1600            1730
2701         48Z55  09 มกราคม 2018           1530            1700
2702         90081  09 มกราคม 2018           1600            1700

**ผู้ต้องสงสัยคนที่ 1**

name : Joe Germuska

person_id : 28819

member_id : 48Z7A

--------------------

**ผู้ต้องสงสัยคนที่ 2**

name : Jeremy Bowers

person_id : 67318

member_id : 48Z55

--------------------
จะเห็นได้ว่า จากคำให้การของคุณ Morty Schapiro ชี้ไปที่ Jeremy Bowers เมื่อนำมาประกอบกับคำให้การของคุณ Annabel Miller ที่พบฆาตรกรที่ยิมเมื่อวันที่ 09 มกราคม 2018 ก็ชี้ไปที่ Jeremy Bowers เช่นกัน

สรุปแล้ว ผู้ต้องสงสัยตอนนี้ คือ Jeremy Bowers

#ข้อมูลผู้ต้องสงสัย

In [30]:
Person_df[Person_df['name'] == 'Jeremy Bowers']

id           name  license_id  address_number    address_street_name  \
6327  67318  Jeremy Bowers      423327             530  Washington Pl, Apt 3A   

            ssn  
6327  871539279

#คำให้การของผู้ต้องสงสัย

In [31]:
Interview_df[Interview_df['person_id'] == 67318]

person_id                                         transcript
4990      67318  I was hired by a woman with a lot of money. I ...

In [32]:
Interview_df[Interview_df['person_id'] == 67318].iloc[0]['transcript']

'I was hired by a woman with a lot of money. I don\'t know her name but I know she\'s around 5\'5" (65") or 5\'7" (67"). She has red hair and she drives a Tesla Model S. I know that she attended the SQL Symphony Concert 3 times in December 2017.\n'

Jeremy Bowers ให้การว่า "ฉันถูกจ้างโดยผู้หญิงคนหนึ่งที่มีเงินมากมาย ฉันไม่ทราบชื่อของเธอ แต่ฉันรู้ว่าเธอสูงประมาณ 5 ฟุต 5 นิ้ว (65 นิ้ว) หรือ 5 ฟุต 7 นิ้ว (67 นิ้ว) เธอมีผมสีแดงและขับรถ Tesla Model S ฉันรู้ว่าเธอไปชมคอนเสิร์ต SQL Symphony 3 ครั้งในเดือนธันวาคม 2017"

สรุปตอนนี้ Jeremy Bowers ถูกจ้างวานให้ฆ่า

##สืบต่อเพื่อหาคนจ้างตามคำให้การของ Jeremy Bowers

In [33]:
filtered_df = DriverLicense_df[
    (DriverLicense_df['hair_color'].str.contains('red')) &
    (DriverLicense_df['car_make'].str.contains('Tesla')) &
    (DriverLicense_df['car_model'].str.contains('S'))
]
filtered_df

id  age  height eye_color hair_color  gender plate_number car_make  \
1105  202298   68      66     green        red  female       500123    Tesla   
2054  291182   65      66      blue        red  female       08CM64    Tesla   
7083  736081   79      69     brown        red    male       GCAQ6Y    Tesla   
9078  918773   48      65     black        red  female       917UU3    Tesla   

     car_model  
1105   Model S  
2054   Model S  
7083   Model S  
9078   Model S

In [34]:
Person_df[Person_df['license_id'].isin(filtered_df['id'])]

id              name  license_id  address_number address_street_name  \
5275  57410    Cletus Zoeller      736081            2987         Kingham Way   
7664  78881          Red Korb      918773             107         Camerata Dr   
8977  90700     Regina George      291182             332           Maple Ave   
9985  99716  Miranda Priestly      202298            1883          Golden Ave   

            ssn  
5275  924648898  
7664  961388910  
8977  337169072  
9985  987756388

In [35]:
#Cletus Zoeller
FacebookEventCheckin_df[FacebookEventCheckin_df['person_id'] == 57410]

person_id  event_id                                         event_name  \
3888      57410      3841                                          harder.\n   
3889      57410      6505  God did not create the world in 7 days; he scr...   
3890      57410      1649                                knows what it is.\n   

          date  
3888  20170610  
3889  20171119  
3890  20170701

In [36]:
#Red Korb
FacebookEventCheckin_df[FacebookEventCheckin_df['person_id'] == 78881]

Empty DataFrame
Columns: [person_id, event_id, event_name, date]
Index: []

In [37]:
#Regina George
FacebookEventCheckin_df[FacebookEventCheckin_df['person_id'] == 90700]

Empty DataFrame
Columns: [person_id, event_id, event_name, date]
Index: []

In [38]:
#Miranda Priestly
FacebookEventCheckin_df[FacebookEventCheckin_df['person_id'] == 99716]

person_id  event_id            event_name      date
20006      99716      1143  SQL Symphony Concert  20171206
20007      99716      1143  SQL Symphony Concert  20171212
20008      99716      1143  SQL Symphony Concert  20171229

จะเห็นได้ว่า Miranda Priestly	ไปชมคอนเสิร์ต SQL Symphony 3 ครั้งในเดือนธันวาคม 2017 จริง ตามคำให้การของ Jeremy Bowers

###ตรวจสอบรายได้

In [39]:
Person_df[Person_df['name'] == 'Miranda Priestly']

id              name  license_id  address_number address_street_name  \
9985  99716  Miranda Priestly      202298            1883          Golden Ave   

            ssn  
9985  987756388

In [40]:
Income_df[Income_df['ssn'] == 987756388]

ssn  annual_income
7422  987756388         310000

In [41]:
Income_df[Income_df['ssn'] == 871539279]

ssn  annual_income
6474  871539279          10500

**สรุป** คดีฆาตรกรรมที่เกิดขึ้นในวันที่ 15 มกราคม 2018 ที่เมือง SQL City
มีผู้กระทำผิด 2 คน ได้แก่

Miranda Priestly ที่เป็นผู้จ้างวานให้ฆ่า และ Jeremy Bowers ผู้เป็นคนที่ลงมือฆ่า